lib/SILGen/SILGenDistributed.cpp

```diff
@@ -494,21 +609,62 @@ void SILGenFunction::emitDistributedActorFactory(FuncDecl *fd) {
   SILValue selfArgValue = F.getSelfArgument();
   ManagedValue selfArg = ManagedValue::forUnmanaged(selfArgValue);
 
-  // ==== Case 'remote') Create the remote instance
+  // type: SpecificDistributedActor.Type
+  auto selfArgType = F.mapTypeIntoContext(selfArg.getType().getASTType());
+  auto selfMetatype = getLoweredType(selfArgType);
+  SILValue selfMetatypeValue = B.createMetatype(loc, selfMetatype);
+
+  // --- Prepare param: Self.self
+  // type: SpecificDistributedActor
+  auto returnTy = getLoweredType(
+      F.mapTypeIntoContext(selfTyDecl->getDeclaredInterfaceType()));
+
+  // ==== Prepare all the basic blocks
+  auto returnBB = createBasicBlock();
+  auto resolvedBB = createBasicBlock();
+  auto makeProxyBB = createBasicBlock();
+  auto switchBB = createBasicBlock();
+  auto errorBB = createBasicBlock();
+
+  SILFunctionConventions fnConv = F.getConventions(); // TODO: no idea?
```

Comment:
```suggestion
  SILFunctionConventions fnConv = F.getConventions(); 
```

seems to be right I guess :)